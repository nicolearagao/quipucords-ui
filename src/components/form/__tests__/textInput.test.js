import React from 'react';
import { TextInputTypes } from '@patternfly/react-core';
import { TextInput } from '../textInput';

describe('TextInput Component', () => {
  it('should render a basic component', () => {
    const props = {};

    const component = renderComponent(<TextInput {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should export support constants, types', () => {
    expect({ TextInputTypes }).toMatchSnapshot('support');
  });

  it('should handle readOnly, disabled', () => {
    const props = {
      isReadOnly: true
    };

    const component = renderComponent(<TextInput {...props} />);
    expect(component).toMatchSnapshot('readOnly');

    const componentDisabled = component.setProps({
      isReadOnly: false,
      isDisabled: true
    });

    expect(componentDisabled).toMatchSnapshot('disabled');

    const componentActive = component.setProps({
      isReadOnly: false,
      isDisabled: false
    });

    expect(componentActive).toMatchSnapshot('active');
  });

  it('should return an emulated onChange event', () => {
    const props = {
      onChange: jest.fn(),
      value: 'lorem ipsum'
    };

    const component = renderComponent(<TextInput {...props} />);
    const input = component.find('input');
    const mockEvent = { target: { value: 'dolor sit' } };
    component.fireEvent.change(input, mockEvent);

    expect(props.onChange.mock.calls).toMatchSnapshot('emulated event, change');
  });

  it('should return an emulated onClear event on escape', () => {
    const props = {
      id: 'test-id',
      value: 'lorem ipsum',
      onKeyUp: jest.fn(),
      onClear: jest.fn()
    };

    const component = renderComponent(<TextInput {...props} />);
    const input = component.find('input');
    const mockEvent = { target: { value: '' }, keyCode: 27, which: 27, key: 'Escape' };
    component.fireEvent.keyUp(input, mockEvent);

    expect(props.onKeyUp).toHaveBeenCalledTimes(1);
    expect(props.onClear).toHaveBeenCalledTimes(1);
    expect(props.onClear.mock.calls).toMatchSnapshot('emulated event, esc');
  });

  it('should return a mouseup event on text clear', () => {
    const props = {
      onMouseUp: jest.fn(),
      value: 'lorem ipsum'
    };

    const component = renderComponent(<TextInput {...props} />);
    const input = component.find('input');
    const mockEvent = { target: { value: '' } };
    component.fireEvent.mouseUp(input, mockEvent);

    expect(props.onMouseUp.mock.calls).toMatchSnapshot('emulated event, mouseup');
  });
});
