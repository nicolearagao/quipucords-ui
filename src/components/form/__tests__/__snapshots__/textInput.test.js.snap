// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component should export support constants, types: support 1`] = `
{
  "TextInputTypes": {
    "date": "date",
    "datetimeLocal": "datetime-local",
    "email": "email",
    "month": "month",
    "number": "number",
    "password": "password",
    "search": "search",
    "tel": "tel",
    "text": "text",
    "time": "time",
    "url": "url",
  },
}
`;

exports[`TextInput Component should handle readOnly, disabled: active 1`] = `
<textinput
  props="{
  "isReadOnly": false,
  "className": "",
  "id": null,
  "isDisabled": false,
  "name": null,
  "type": "text",
  "value": "",
  "children": []
}"
>
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    data-ouia-component-id="OUIA-Generated-TextInputBase-4"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="generatedid-"
    name="generatedid-"
    type="text"
    value=""
  />
</textinput>
`;

exports[`TextInput Component should handle readOnly, disabled: disabled 1`] = `
<textinput
  props="{
  "isReadOnly": false,
  "className": "",
  "id": null,
  "isDisabled": true,
  "name": null,
  "type": "text",
  "value": "",
  "children": []
}"
>
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    data-ouia-component-id="OUIA-Generated-TextInputBase-3"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    disabled=""
    id="generatedid-"
    name="generatedid-"
    type="text"
    value=""
  />
</textinput>
`;

exports[`TextInput Component should handle readOnly, disabled: readOnly 1`] = `
<textinput
  props="{
  "isReadOnly": true,
  "className": "",
  "id": null,
  "isDisabled": false,
  "name": null,
  "type": "text",
  "value": "",
  "children": []
}"
>
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    data-ouia-component-id="OUIA-Generated-TextInputBase-2"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="generatedid-"
    name="generatedid-"
    readonly=""
    type="text"
    value=""
  />
</textinput>
`;

exports[`TextInput Component should render a basic component: basic 1`] = `
<textinput
  props="{
  "className": "",
  "id": null,
  "isDisabled": false,
  "isReadOnly": false,
  "name": null,
  "type": "text",
  "value": "",
  "children": []
}"
>
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    data-ouia-component-id="OUIA-Generated-TextInputBase-1"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="generatedid-"
    name="generatedid-"
    type="text"
    value=""
  />
</textinput>
`;

exports[`TextInput Component should return a mouseup event on text clear: emulated event, mouseup 1`] = `
[
  [
    {
      "checked": undefined,
      "currentTarget": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-7"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="text"
        value="lorem ipsum"
      />,
      "id": "generatedid-",
      "keyCode": undefined,
      "name": "generatedid-",
      "persist": [Function],
      "target": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-7"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="text"
        value="lorem ipsum"
      />,
      "value": "",
    },
  ],
]
`;

exports[`TextInput Component should return an emulated onChange event: emulated event, change 1`] = `
[
  [
    {
      "checked": undefined,
      "currentTarget": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-5"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="text"
        value="dolor sit"
      />,
      "id": "generatedid-",
      "keyCode": undefined,
      "name": "generatedid-",
      "persist": [Function],
      "target": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-5"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="text"
        value="dolor sit"
      />,
      "value": "dolor sit",
    },
  ],
]
`;

exports[`TextInput Component should return an emulated onClear event on escape: emulated event, esc 1`] = `
[
  [
    {
      "checked": undefined,
      "currentTarget": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-6"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="test-id"
        name="generatedid-"
        type="text"
        value=""
      />,
      "id": "test-id",
      "keyCode": 27,
      "name": "generatedid-",
      "persist": [Function],
      "target": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-6"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="test-id"
        name="generatedid-"
        type="text"
        value=""
      />,
      "value": "",
    },
  ],
]
`;
